### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Daniel-Kigozi**
In this milestone, I did the following
- Add devise for authentication

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (VICTOR-CHUKWU) [inline app/models/user.rb]: - [ ] a user should be able to confirm his email, kindly add `:confirmable` to devise

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users
```
Comment (VICTOR-CHUKWU) [inline config/routes.rb]: - [ ] an unauthorized user should be redirected to the sign_up or sign_in page, not the user index page

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Daniel-Kigozi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — COMMENTED** [review]
## Project Status: INVALIDATED :red_circle: 
Hi @Daniel-Kigozi , 

While you made a great effort in this project, I can't go ahead and review your code.


## Invalid Code Review Request

- This is because you have submitted a project, that has an error that is preventing your app from running and I think its because of introducing the authenticated and unauthenticated routes in your `routes.rb` file so kindly fix this :+1: 
- Error Message
image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Daniel-Kigozi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**